sympy/matrices/common.py

```diff
@@ -2393,6 +2393,27 @@ def __pow__(self, exp):
 
         return self.pow(exp)
 
+    def matrix_exp(self, exp, side = "left"):
```

Comment:
I think that the name is confusing since there is already `exp`
Better candidates are `exp_by_matrix`, `pow_by_matrix`, `mat_mat_pow`

Reply:
I think it would be better to have different functions for the left and right matrix powers.

It would be good to have these as symbolic operations for matrix expressions.

Reply:
```suggestion
    def matrix_exp(self, exp, side="left"):
```